### PR TITLE
feat: write unit tests to show the regex seems way too permissive

### PR DESCRIPTION
**Pull request by danielgent**
I really think we should remove the regex (as I complained ages ago :smile:)  It's an exploit waiting to happen

I could write a util function that matches `*` for sub domains. If I don't hear back from you I'll just do it :sunglasses: 

*also* @markusahlstrand have I shown you https://jestjs.io/docs/api#describeeachtablename-fn-timeout ?

I can find some examples in the other projects

We could have a table with `test name, valid URLs, input, expected output` and it would be much more readable. WIth this many small tests I lose track :sunglasses: 

---------------------------------------------------------------------------------------

Thinking about this PR, it's patching what I see as too permissive allowed callback URLs
Only we enter these, not clients

*if I had proved that the URLs we are validating could bypass our protection then that would be an issue* but it's not really

I still think we should do this though

image

### CODE REVIEW

File: test/utils/validate-redirect-url.spec.ts [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+
+  it("should allow wildcard like this?", () => {
+    const allowedUrls = ["https://*"];
+    const redirectUri = "https://*";
+    expect(() => validateRedirectUrl(allowedUrls, redirectUri)).not.toThrow();
+  });
+
+  it("really should not allow this", () => {
+    const allowedUrls = ["*"];
+    const redirectUri = "*";
```
Comment (danielgent) [inline test/utils/validate-redirect-url.spec.ts:36]: :smile: vive le testing!

**Comment by markusahlstrand** [comment]
Hmm.. this does indeed look like an issue..

**Review by markusahlstrand — APPROVED** [review]
Hard to argue with passing tests..

**Comment by danielgent** [comment]
> Hard to argue with passing tests..

Indeed. It's this one that you may have comments, questions, and concerns over! :smile:

**Pull request merged**